sdk/core/Azure.Core.TestFramework/src/ProxyTransport.cs

```diff
@@ -137,17 +149,14 @@ private void RedirectToTestProxy(HttpMessage message)
             request.Headers.SetValue("x-recording-id", _recording.RecordingId);
             request.Headers.SetValue("x-recording-mode", _recording.Mode.ToString().ToLower());
 
-            // Ensure x-recording-upstream-base-uri header is only set once, since the same HttpMessage will be reused on retries
-            if (!request.Headers.Contains("x-recording-upstream-base-uri"))
+            // Intentionally reset the upstream URI in case the request URI changes between retries - e.g. when using GeoRedundant secondary Storage
+            var baseUri = new RequestUriBuilder()
             {
-                var baseUri = new RequestUriBuilder()
-                {
-                    Scheme = request.Uri.Scheme,
-                    Host = request.Uri.Host,
-                    Port = request.Uri.Port,
-                };
-                request.Headers.SetValue("x-recording-upstream-base-uri", baseUri.ToString());
-            }
+                Scheme = request.Uri.Scheme,
+                Host = request.Uri.Host,
+                Port = request.Uri.Port,
+            };
+            request.Headers.SetValue("x-recording-upstream-base-uri", baseUri.ToString());
```

Comment:
Should we make all these string literals into constants somewhere?

Reply:
I'm not a huge fan of doing that if they are not reused.